compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java

```diff
@@ -269,6 +274,10 @@ private SymbolTable(CompilerContext context) {
         this.trueType = new BFiniteType(finiteTypeSymbol, new HashSet<>() {{
             add(trueLiteral);
         }});
+        this.anyAndReadonly =
+                ImmutableTypeCloner.getImmutableIntersectionType((SelectivelyImmutableReferenceType) this.anyType,
+                        this.symbolTable, names);
+        initializeType(this.anyAndReadonly, TypeKind.ANYANDREADONLY.typeName(), BUILTIN);
```

Comment:
Instead of introducing a `TypeKind` just for this, I think we can extract the name from the `effectiveType` of the `anyAndReadonly` BIntersectionType.

Reply:
Changed.